qutip/core/dimensions.py

```diff
@@ -365,17 +383,37 @@ def __call__(cls, *args: SpaceLike, rep: str = None) -> "Space":
         if cls is SuperSpace:
             args = (*args, rep or 'super')
 
-        if args not in cls._stored_dims:
+        if cls is SumSpace:
+            if len(args) > 1 and all(arg == args[0] for arg in args):
+                repeat = len(args)
+                args = (args[0],)
+            kwargs = {'repeat': repeat}
+            stored_args = (*args, *kwargs.items())
+        else:
+            kwargs = {}
+            stored_args = args
```

Comment:
`kwargs` and `stored_args` feels over complicated...
If you change the SumSpace.__init__ signature to `(self, spaces: tuple[Space], repeat=None)` then `args = (args, repeat)`, no need for new variables.
You could also remove the `repeat` and create the _homtuple here.